lib/vm/src/instance.rs

```diff
@@ -761,99 +680,98 @@ impl Instance {
     }
 }
 
-/// A handle holding an `Instance` of a WebAssembly module.
-#[derive(Hash, PartialEq, Eq)]
-pub struct InstanceHandle {
-    instance: *mut Instance,
-
-    /// Whether `Self` owns `self.instance`. It's not always the case;
-    /// e.g. when `Self` is built with `Self::from_vmctx`.
-    ///
-    /// This information is necessary to know whether `Self` should
-    /// deallocate `self.instance`.
-    owned_instance: bool,
+#[derive(Debug)]
+#[repr(C)]
+pub struct InstanceAllocator {
+    ptr_to_dealloc: *const Arc<Self>,
```

Comment:
Confused about the `Arc<Self>` here

Reply:
I'll add very verbose comment to explain everything here :-).